packages/hub/node-tests/routes/scheduled-payments-test.ts

```diff
@@ -0,0 +1,307 @@
+            },
+            status: '422',
+            title: 'Invalid attribute',
+          },
+        ],
+      });
+  });
+
+  // TODO add a test for recurring scheduled payment
+
+  it('persists scheduled payment, submits a transaction and transaction waiter task', async function () {
+    let scheduledPaymentId;
+    await request()
+      .post('/api/scheduled-payments')
+      .send({
+        data: {
+          type: 'scheduled-payments',
+          attributes: {
+            'sender-safe-address': '0xc0ffee254729296a45a3885639AC7E10F9d54979',
+            'module-address': '0x7E7d0B97D663e268bB403eb4d72f7C0C7650a6dd',
+            'token-address': '0xa455bbB2A81E09E0337c13326BBb302Cb37D7cf6',
+            amount: 100,
+            'payee-address': '0x821f3Ee0FbE6D1aCDAC160b5d120390Fb8D2e9d3',
+            'execution-gas-estimation': 100000,
+            'max-gas-price': 1000000000,
+            'fee-fixed-usd': 0,
+            'fee-percentage': 0,
+            salt: '54lt',
+            'pay-at': '2021-01-01T00:00:00.000Z',
+            'sp-hash': '0x123',
+            'chain-id': 1,
+            signature: '0x123',
+          },
+        },
+      })
+      .set('Accept', 'application/vnd.api+json')
+      .set('Authorization', 'Bearer abc123--def456--ghi789')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(201)
+      .expect(function (res) {
+        scheduledPaymentId = res.body.data.id;
+        res.body.data.id = 'id';
+      })
+      .expect({
+        data: {
+          id: 'id',
+          type: 'scheduled-payment',
+          attributes: {
+            'sender-safe-address': '0xc0ffee254729296a45a3885639AC7E10F9d54979',
+            'module-address': '0x7E7d0B97D663e268bB403eb4d72f7C0C7650a6dd',
+            'token-address': '0xa455bbB2A81E09E0337c13326BBb302Cb37D7cf6',
+            amount: 100,
+            'payee-address': '0x821f3Ee0FbE6D1aCDAC160b5d120390Fb8D2e9d3',
+            'execution-gas-estimation': 100000,
+            'max-gas-price': 1000000000,
+            'fee-fixed-usd': 0,
+            'fee-percentage': 0,
+            salt: '54lt',
+            'pay-at': '2021-01-01T00:00:00.000Z',
+            'sp-hash': '0x123',
+            'chain-id': 1,
+            signature: '0x123',
+            'creation-transaction-hash': mockTxnHash,
+            'creation-block-number': null,
+            'cancelation-transaction-hash': null,
+            'cancelation-block-number': null,
+            'recurring-day-of-month': null,
+            'recurring-until': null,
+          },
+        },
+      })
+      .expect('Content-Type', 'application/vnd.api+json');
+
+    expect(getJobIdentifiers()[0]).to.equal('scheduled-payment-on-chain-creation-waiter');
+    expect(getJobPayloads()[0]).to.deep.equal({ scheduledPaymentId });
+  });
+
+  it('does not persist a scheduled payment when something goes wrong with on chain call', async function () {
+    forceSchedulePaymentOnChainTimeout = true;
```

Comment:
Maybe this should be reset to `false` in a `beforeEach` in case any tests get added after this?

Reply:
Sounds good. Will change